qa/rpc-tests/dip3-deterministicmns.py

```diff
@@ -0,0 +1,726 @@
+
+    def start_alias(self, node, alias, should_fail=False):
+        start_result = node.masternode('start-alias', alias)
+        if not should_fail:
+            assert_equal(start_result, {'result': 'successful', 'alias': alias})
+        else:
+            assert_equal(start_result, {'result': 'failed', 'alias': alias, 'errorMessage': 'Failed to verify MNB'})
+
+    def generate_blocks_until_winners(self, node, count, timeout=60):
+        # Winner lists are pretty much messed up when too many blocks were generated in a short time
+        # To allow proper testing of winners list, we need to slowly generate a few blocks until the list stabilizes
+        good_count = 0
+        st = time.time()
+        while time.time() < st + timeout:
+            height = node.getblockchaininfo()['blocks'] + 10
+            winners = node.masternode('winners')
+            if str(height) in winners:
+                if re.match('[0-9a-zA-Z]*:10', winners[str(height)]):
+                    good_count += 1
+                    if good_count >= count:
+                        return
+                else:
+                    good_count = 0
+            node.generate(1)
+            self.sync_all()
+            time.sleep(1)
+        raise AssertionError("generate_blocks_until_winners timed out: {}".format(node.masternode('winners')))
+
+    def test_mn_votes(self, block_count, test_enforcement=False):
+        self.generate_blocks_until_winners(self.nodes[0], self.num_nodes)
+
+        if test_enforcement:
+            self.nodes[0].spork('SPORK_8_MASTERNODE_PAYMENT_ENFORCEMENT', 0)
+            self.wait_for_sporks()
+            self.test_invalid_mn_payment(self.nodes[0])
+
+        cur_block = 0
+        while cur_block < block_count:
+            for n1 in self.nodes:
+                if cur_block >= block_count:
+                    break
+                if n1 is None:
+                    continue
+
+                if test_enforcement:
+                    self.test_invalid_mn_payment(n1)
+
+                n1.generate(1)
+                cur_block += 1
+                self.sync_all()
+
+                height = n1.getblockchaininfo()['blocks']
+                winners = self.wait_for_winners(n1, height + 10)
+
+                for n2 in self.nodes:
+                    if n1 is n2 or n2 is None:
+                        continue
+                    winners2 = self.wait_for_winners(n2, height + 10)
+                    if winners[str(height + 10)] != winners2[str(height + 10)]:
+                        print("winner1: " + str(winners[str(height + 10)]))
+                        print("winner2: " + str(winners2[str(height + 10)]))
+                        raise AssertionError("winners did not match")
+
+        if test_enforcement:
+            self.nodes[0].spork('SPORK_8_MASTERNODE_PAYMENT_ENFORCEMENT', 4070908800)
+
+    def test_instantsend(self, tx_count, repeat):
+        self.nodes[0].spork('SPORK_2_INSTANTSEND_ENABLED', 10000)
```

Comment:
Why `10000`?

Reply:
Good question...changed to 0. It might be because at some point in time I was wondering if sporks are propagated and then used 10000 instead...